### PR TITLE
Wait for entities to load when the a-node has not been yet initialized (fix #2802)

### LINKED ISSUE #2802: Nested entities initialize out of order in scene without a-assets
**Description:**

I believe components can assume during their initialization that components on child nodes have already been initialized due to the inside-out initialization order. This assumption does not hold up in a scene with no `<a-assets>` declared. 

- A-Frame Version: 0.5.0
- Platform / Device: Win10, Chome, Chromium, Firefox, & Nightly
- Reproducible Code Snippet or URL:

This scene errors when a component on `<a-locomotor>` expects an object to be returned from `getAttribute` on a child node component but gets a string instead: 

```html
<head>
  <title>Most Basic Super-Hands Example</title>
  <script src="https://aframe.io/releases/0.5.0/aframe.min.js"></script>
  <script src="//cdn.rawgit.com/donmccurdy/aframe-extras/v3.3.0/dist/aframe-extras.min.js"></script>
  <script src="https://rawgit.com/wmurphyrd/aframe-super-hands-component/master/dist/super-hands.js"></script>
</head>

<body>
  <a-scene>
    <a-locomotor>
      <!-- Make sure your super-hands entities also have controller and collider components -->
      <a-entity hand-controls="left" super-hands sphere-collider="objects: a-box"></a-entity>
      <a-entity hand-controls="right" super-hands sphere-collider="objects: a-box"></a-entity>
    </a-locomotor>
    <!-- hover & drag-drop won't have any obvious effect without some additional event handlers or components. See the examples page for more -->
    <a-box hoverable grabbable stretchable drag-droppable
    color="blue" position="0 0 -1"></a-box>
  </a-scene>
</body>
```

This scene, identical except for the addition of an empty `<a-assets>` loads successfully:

```html
<head>
  <title>Most Basic Super-Hands Example</title>
  <script src="https://aframe.io/releases/0.5.0/aframe.min.js"></script>
  <script src="//cdn.rawgit.com/donmccurdy/aframe-extras/v3.3.0/dist/aframe-extras.min.js"></script>
  <script src="https://rawgit.com/wmurphyrd/aframe-super-hands-component/master/dist/super-hands.js"></script>
</head>

<body>
  <a-scene>
    <a-assets></a-assets>
    <a-locomotor>
      <!-- Make sure your super-hands entities also have controller and collider components -->
      <a-entity hand-controls="left" super-hands sphere-collider="objects: a-box"></a-entity>
      <a-entity hand-controls="right" super-hands sphere-collider="objects: a-box"></a-entity>
    </a-locomotor>
    <!-- hover & drag-drop won't have any obvious effect without some additional event handlers or components. See the examples page for more -->
    <a-box hoverable grabbable stretchable drag-droppable
    color="blue" position="0 0 -1"></a-box>
  </a-scene>
</body>
```

This can definitely be solved in the component by not making the assumption, but I thought it was worth documenting for you.

### PR DESCRIPTION
**Pull request by dmarcos**
It checks for the tagName to be `a-node` , `a-entity` or a registered primitive.

### CODE REVIEW

**Comment by dmarcos** [comment]
We should make sure there are no other `a-nodes` the parent should be waiting for? `a-animation`, `a-asseet-item`... Not sure if `a-assets` need some rethinking to make sure we block rendering on them

File: src/core/a-node.js [hunk 1]
```diff
@@ -97,11 +97,16 @@ module.exports = registerElement('a-node', {
...
         var childrenLoaded;
         var self = this;
+        var primitives = AFRAME.primitives.primitives;
 
         if (this.hasLoaded) { return; }
 
         // Default to waiting for all nodes.
-        childFilter = childFilter || function (el) { return el.isNode; };
+        childFilter = childFilter || function (el) {
+          var tagName = el.tagName.toLowerCase();
```
Comment (ngokevin) [inline src/core/a-node.js]: The prototype is not available?

File: src/core/a-node.js [hunk 2]
```diff
@@ -97,11 +97,16 @@ module.exports = registerElement('a-node', {
...
+          var isEntity = tagName === 'a-entity' || tagName === 'a-node' || primitives[tagName] !== undefined;
+          return el.isNode || isEntity;
```
Comment (ngokevin) [inline src/core/a-node.js]: This function should check isNode the very first. If not, only the execute the lowerCase and other checks

**Comment by dmarcos** [comment]
The prototype of the children elements is not set by the time `createdCallback` and `attachCallback` are called on the parent. I'm using the existing method `isNode` now that checks for the registered tags (https://github.com/aframevr/aframe/blob/master/src/core/a-register-element.js#L32). I think this should be sufficient.

File: tests/core/a-entity.test.js [hunk 3]
```diff
@@ -641,6 +643,52 @@ suite('a-entity', function () {
         done();
       });
     });
+
+    test('wait for all the node children that are not yet nodes to load', function (done) {
```
Comment (ngokevin) [inline tests/core/a-entity.test.js]: These tests pass for me without the changes. Do these reproduce the issue?

**Comment by dmarcos** [comment]
I had run the tests with breakpoints that introduce changes in the timing. I added extra conditions to the tests so they're checking what they're supposed to test: The parent waits for the children nodes to load when the `isNode` flag is not set yet (because `createdCallback` has not been yet invoked)

**Comment by ngokevin** [comment]
Need to remove the `.only`.

**Comment by ngokevin** [comment]
r+wc nice, i forgot we did that `isNode` function

**Pull request merged**